data_structures/MIddleElement_S.ll.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file 
+ * @brief Implementation of [Finding middle element in 
+ * a single linked list](https://simple.wikipedia.org/wiki/Linked_list)
+ * @details
+ * The linked list is a data structure used for holding a sequence of
+ * values, which can be added, displayed, reversed, or removed.
+ * ### Algorithm
+ * This can be done by the slow and fast pointer approach 
+ * (two pointer approach),basically in this we make two pointer
+ * slow and fast we point slow = head , and fast to the head->next
+ * and if length of the L.L is even then traverse till fast!=NULL
+ * and we move slow by one step and fast by one step and when fast
+ * became null then we just print the slow->data , (our mid element).
+ * For the odd length all the process will be same but only one diff. 
+ * isthat we need to check instead of fast!=null we have to chek 
+ * fast->next!=null and similarlly we print the slow->data
+ */
+#include <iostream> /// for I/O operations
+
+/**
+ * @brief A Node class containing a value and pointer to another link
+ * Containing the explicit constructor which contains
+ * after creating node and next to nullptr. 
+ */
+class node{
+public:
+    int data=0;            ///< The value / key of the node 
+    node*next;          ///<  pointer pointing to the next node adress.
+ /*
+ * @param data stores the value of the node
+ */
+    explicit node(int data){
+        this->data=data;
+        next=nullptr;
+    }
+};
+
+/**
+ * @brief Function use to calculate the mid element of the linked list
+ * @param len length of the linked list
+ * @param head stores the first node address of the LinkedList
+ * @returns void 
+ */
+void midpoint(int len,node*head){
+    node*slow = head;
+    node*fast=head->next;
+    if(len%2){
+        while(fast!=nullptr){
+            slow=slow->next;
+            fast=fast->next->next;
+        }
+        std::cout<<slow->data<<" ";
+    }
+    else{
+        while(fast->next!=nullptr){
+            slow=slow->next;
+            fast=fast->next->next;
+        }
+        std::cout<<slow->data<<" ";
+    }
+}
+
+/**
+ * @brief function takes the head node and traverses the l.l
+ * and prints the data 
+ * @param head stores the first node address of the linkedlist 
+ * @returns void 
+ */
+void print(node*head){
+    node*temp = head;
+    while(temp!=nullptr){
+        std::cout<<temp->data<<" ";
+        temp = temp->next;
+    }
+}
+
+/**
+ * @brief Utility function to calculate the length of the linked list
+ * @param head stores the first node address of the LinkedList
+ * @returns the length of the linked list
+ */
+int length(node*head){
+    int len =0;
+    node*temp = head;
+    while(temp!=nullptr){
+        temp=temp->next;
+        len++;
+    }
+    std::cout<<"The middle element is:";
+    return len;
+    /*
+    * returning the final length of the linked list 
+    */
+}
+
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃